### ISSUE TITLE
failed in Integrating 1 million cells

### ISSUE THREAD

**Issue by screamer** [brianhie/scanorama#69]
I used the scripts in ./bin/mouse_brain.py

  BATCH_SIZE = 10000
  t0 = time()
    datasets_dimred, genes = integrate(
        datasets, genes_list, ds_names=data_names,
        batch_size=BATCH_SIZE,
    )
    print('Integrated panoramas in {:.3f}s'.format(time() - t0))

    t0 = time()
    datasets_dimred, datasets, genes = correct(
        datasets, genes_list, ds_names=data_names,
        return_dimred=True, batch_size=BATCH_SIZE,
    )

The error looks like this:
-packages/scipy/sparse/compressed.py", line 88, in __init__
    self._set_self(self.__class__(coo_matrix(arg1, dtype=dtype)))
  File "/home/jialu/packages/anaconda3/envs/scanoramaEnv/lib/python3.7/site-packages/scipy/sparse/coo.py", line 192, in __init__
    self.data = M[self.row, self.col]
MemoryError: Unable to allocate array with shape (2123370384,) and data type float64

The machine has 256G memory. It should be enough for running Scanorama on this data, right?

**Comment by brianhie** [comment]
Hi @screamer, is the MemoryError happening in the correct() function? You can take a look at the whole stack trace to make sure.

You might also try decreasing the batch size. I believe 256GB should be sufficient.

You can also try sketching-based integration for additional efficiency: https://github.com/brianhie/scanorama/blob/master/scanorama/scanorama.py#L147

Will close since this doesn't appear to be a bug with Scanorama, but please feel free to ask additional questions!

**Comment by screamer** [comment]
Hello @brianhie , thanks for your fast replying. I used BATCH_SIZE=2000. It returns the same problem. It seems that the error was happening in the integrate() function, because I didn't see the results of print('Integrated panoramas in {:.3f}s'.format(time() - t0)). Could I solved it by use a smaller BATCH_SIZE like 500? or better to use the sketch option? It costs about 18 hours to test the error everytime.
